### ISSUE TITLE
Investigate possible improvements to Docker Swarm deployment

### ISSUE THREAD

**Issue by janmiderback** [qlik-oss/core-assisted-prescription#373]
Assisted Prescription (AP) allows for deployment to Docker Swarm, both in a local set up, where the swarm is provided by the developer, and to AWS where the production AP site is hosted. The CircleCI configuration uses a mix of its config and shell scripts to do the production deployment as part of the build.

It would be good to investigate if the following problems can be addressed:

- The combination of actions made in `config.yml` and `swarm.sh` is very hard to follow. Can this be simplified, preferably not doing less work in the CCI config?
- Actions run in `config.yml` only apply to the `master` branch, so we cannot detect problems before deployment to production (merge to master). Can this be done in some way without introducing a staging env? Can we test on swarm on CCI?

The goal is mainly _simplification_, _reproducibility_, and _deployment confidence_.

**Comment by sublibra** [comment]
Closing this since there is little focus on Docker swarm in the community.
